src/Development/IDE/Core/Compile.hs

```diff
@@ -82,19 +82,22 @@ computePackageDeps env pkg = do
 
 -- | Typecheck a single module using the supplied dependencies and packages.
 typecheckModule
-    :: HscEnv
+    :: Bool
```

Comment:
Putting it as the first parameter is very reasonable but it would be great if we could get a newtype for this to get something a bit more informative than `Bool`.